### PR TITLE
docs: add meta descriptions

### PR DESCRIPTION
**Pull request by tara-det-ai**
for compliance and SEO purposes, 

- add meta descriptions to main index pages, api user guide pages, main reference pages, tutorials
- add alt tags to images

Tests
- [x] each meta description should strive to meet 150-170 characters
- [x] each meta description represents the content on the page
- [x] each meta description is unique
- [x] each description includes a call to action where possible

### CODE REVIEW

File: docs/training/apis-howto/api-keras-ug.rst [hunk 1]
```diff
@@ -2,6 +2,9 @@
  Keras API
 ###########
 
+.. meta::
+   :description: Learn how to use the Keras API to train a Keras model. This user guide walks you through loading your data, defining the model, customizing how the model.fit function is called, checkpointing, and callbacks.
```
Comment (drh-determined-ai) [inline docs/training/apis-howto/api-keras-ug.rst:6]: Can `model.fit` but put in double-tildes? I don't know if that's possible or if it matters in a `.. meta::` block.

**Review by drh-determined-ai — APPROVED** [review]
Looks good! I just have a few suggestions.

File: docs/training/apis-howto/deepspeed/deepspeed.rst [hunk 2]
```diff
@@ -4,7 +4,10 @@
  Usage Guide
 #############
 
-This document introduces DeepSpeed and guides you through how to train a PyTorch model with the
+.. meta::
+   :description: This comprehensive guide explains the DeepSpeed API and provides step-by-step instructions for getting started. Learn how to use DeepSpeed API by training a PyTorch model with the DeepSpeed engine.
```
[the remark this answers is not in the document]
Comment (tara-det-ai) [reply docs/training/apis-howto/deepspeed/deepspeed.rst:8]: resolved

File: docs/training/apis-howto/api-keras-ug.rst (hunk 1 above)
Comment (tara-det-ai) [reply docs/training/apis-howto/api-keras-ug.rst:6]: the .. meta:: directive doesn't support inline formatting and Google wouldn't display it even if we used OpenGraph to format it

**Pull request merged**